controller/share_manager_controller.go

```diff
@@ -694,7 +787,28 @@ func (c *ShareManagerController) syncShareManagerPod(sm *longhorn.ShareManager)
 	}
 
 	// If the node where the pod is running on become defective, we clean up the pod by setting sm.Status.State to STOPPED or ERROR
-	// A new pod will be recreated by the share manager controller.
+	// A new pod will be recreated by the share manager controller.  We might get an early warning of that by the pod going stale.
+	isStale, _, err := c.isShareManagerPodStale(sm)
+	if err != nil {
+		log.WithError(err).Warnf("Failed to check isShareManagerPodStale(%v) when syncShareManagerPod", sm.Name)
+	} else if isStale {
+		log.Infof("ShareManager Pod %v is stale", pod.Name)
+	}
+	if isStale {
+		// if we just transitioned to the starting state, while the prior cleanup is still in progress we will switch to error state
+		// which will lead to a bad loop of starting (new workload) -> error (remount) -> stopped (cleanup sm)
```

Comment:
Could you explain a bit more? I am not quite understand how we stuck in the bad loop. Thank you

Reply:
That comment pre-dates me.  It was added in PR https://github.com/longhorn/longhorn-manager/pull/1034

Reply:
I see. Thank you. Let's keep it for now